ccm-connector/src/main/java/com/sequenceiq/cloudbreak/ccmimpl/cloudinit/DefaultCcmV2ParameterSupplier.java

```diff
@@ -30,12 +30,10 @@
 
     public CcmV2Parameters getCcmV2Parameters(@Nonnull String accountId, @Nonnull Optional<String> environmentCrnOpt,
         @Nonnull String clusterGatewayDomain, @Nonnull String agentKeyId) {
-        String requestId = MDCBuilder.getOrGenerateRequestId();
 
-        InvertingProxy invertingProxy = ccmV2Client.awaitReadyInvertingProxyForAccount(requestId, accountId);
-        InvertingProxyAgent invertingProxyAgent = ccmV2Client.registerInvertingProxyAgent(requestId, accountId, environmentCrnOpt,
-                clusterGatewayDomain, agentKeyId);
-        validateCcmV2ConfigResponse(invertingProxy, invertingProxyAgent);
+        InvertingProxyAndAgent invertingProxyAndAgent = getInvertingProxyAndAgent(accountId, environmentCrnOpt, clusterGatewayDomain, agentKeyId);
+        InvertingProxy invertingProxy = invertingProxyAndAgent.getInvertingProxy();
+        InvertingProxyAgent invertingProxyAgent = invertingProxyAndAgent.getInvertingProxyAgent();
 
         LOGGER.debug("CcmV2Config successfully retrieved InvertingProxyHost: '{}', InvertingProxyStatus: '{}', InvertingProxyAgentCrn: '{}', " +
                         "EnvironmentCrnOpt: '{}'", invertingProxy.getHostname(), invertingProxy.getStatus(),
```

Comment:
Same as for `DefaultCcmV2JumpgateParameterSupplier`: What is the purpose of wrapping the `environmentCrn` in an `Optional`? If the CRN is never null, it could be directly printed, otherwise `Optional.of()` will fail.